src/TopoMojo.Hypervisor/Proxmox/ProxmoxVlanManager.cs

```diff
@@ -140,6 +155,60 @@ public async Task<IEnumerable<PveVnet>> Provision(IEnumerable<string> vnetNames)
         public string ResolvePveNetName(string topoName)
             => IsReserved(topoName) ? topoName : _nameService.ToPveName(topoName);
 
+        public async Task Initialize()
+        {
+            _logger.LogDebug($"initializing nets");
+
+            var vnets = await _vnetsApi.GetVnets();
+
+            foreach (var vnet in vnets)
+            {
+                _logger.LogDebug($"Adding to recent vnet cache: {vnet.Alias}");
+                AddToRecentCache(vnet);
+            }
+        }
+
+        public async Task Clean(ConcurrentDictionary<string, Vm> vmCache, string tag = null)
+        {
+            _logger.LogDebug($"cleaning nets [{tag}]");
+
+            var vnets = await _vnetsApi.GetVnets();
+            var vnetsToDelete = new List<string>();
+
+            if (!string.IsNullOrEmpty(tag))
+            {
+                vnets = vnets.Where(x => _nameService.FromPveName(x.Alias).Tag() == tag);
+            }
+
+            // exclude non-tagged
+            vnets = vnets.Where(x => _nameService.FromPveName(x.Alias).Contains('#'));
```

Comment:
Here, you want to exclude basically the reserved "global" nets, right? I'd probably use `.Where(x => !IsReserved(x.Alias)` or something

Reply:
I think anything without that # should be avoided as it may have been created by something other than Topo.